### PR TITLE
PWN-4270 - Gateway API and register_wallet method

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4270
https://p2pvalidator.atlassian.net/wiki/spaces/Wallet/pages/2407202817/API+Gateway+GW+Onboarding
## Description of Work

- start integration with `register_wallet` method

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/infrastructure/network/NetworkModule.kt [hunk 1]
```diff
@@ -90,6 +92,15 @@ object NetworkModule : InjectionModule {
                 interceptor = null
             )
         }
+
+        single(named(FACADE_SERVICE_RETROFIT_QUALIFIER)) {
+            getRetrofit(
+                // TODO
+                baseUrl = "https://replace_later.com",
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/infrastructure/network/NetworkModule.kt:99]: no url yet

File: app/src/main/java/org/p2p/wallet/infrastructure/network/NetworkModule.kt [hunk 2]
```diff
@@ -90,6 +92,15 @@ object NetworkModule : InjectionModule {
                 interceptor = null
             )
         }
+
+        single(named(FACADE_SERVICE_RETROFIT_QUALIFIER)) {
+            getRetrofit(
+                // TODO
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/infrastructure/network/NetworkModule.kt:98]: it's better o place number of task here

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/infrastructure/network/NetworkModule.kt (hunk 2 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/infrastructure/network/NetworkModule.kt:98]: sounds reasonable, will do

**Review by kuznetsov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/gateway/api/request/RegisterWalletRequest.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    class Params(
+        /**
+         * Solana pubkey base58 encoded
+         */
+        @SerializedName("client_id")
+        val clientPublicKeyB58: String,
+        @SerializedName("ethereum_id")
+        val etheriumPublicKeyB58: String,
+        /**
+         * no “+”, E.164 format
```
Comment (btolfa) [inline app/src/main/java/org/p2p/wallet/auth/gateway/api/request/RegisterWalletRequest.kt:19]: actually with "+"

File: app/src/main/java/org/p2p/wallet/auth/gateway/api/request/RegisterWalletRequest.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    val params: Params
+) {
+    class Params(
+        /**
+         * Solana pubkey base58 encoded
+         */
+        @SerializedName("client_id")
+        val clientPublicKeyB58: String,
+        @SerializedName("ethereum_id")
+        val etheriumPublicKeyB58: String,
```
Comment (btolfa) [inline app/src/main/java/org/p2p/wallet/auth/gateway/api/request/RegisterWalletRequest.kt:17]: etherium address is a hex of the first 20 bytes of etherium pubkey, not a base58 (like solana pubkey)

**Pull request merged**